_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * This is the base class for every Core API endpoint Jetpack uses.
+ *
+ */
+class Jetpack_Core_API_Module_Activate_Endpoint
+	extends Jetpack_Core_API_XMLRPC_Consumer_Endpoint
+	implements Jetpack_Core_API_Writable {
+
+	private $modules_requiring_public = array( 'sitemaps', 'photon', 'enhanced-distribution' );
```

Comment:
Not sure how you determined which blogs required public visibility, but to a lesser extent, `sharing` [requires public](https://github.com/Automattic/jetpack/blob/8d13e90a72b1afad48a6d3c380019c4834c922f3/modules/sharedaddy/sharing.php#L212). Maybe it's not needed for you, but wanted to point it out just in case.